packages/tx/src/EIP2930Transaction.ts

```diff
@@ -172,7 +172,7 @@ export class EIP2930Transaction extends BaseTransaction<EIP2930Transaction> {
 
     this.chainId = chainId ? new BN(toBuffer(chainId)) : new BN(this.common.chainId())
     this.accessList = usedAccessList
-    this.v = v ? new BN(v) : undefined
+    this.v = v ? new BN(toBuffer(v)) : undefined
```

Comment:
This is fine, just as a note here; I did not use `toBuffer` this because `v` can only be 0 or 1.

Reply:
Just as a note: this was nevertheless not done just for cosmetics, since otherwise the `fromTxData()` constructor would break due to allowing these `BNLike` values as an input. This would then break for a `string` input (as happened for my test data).